### ISSUE TITLE
[FEATURE] Internal change for the handling of external data

### ISSUE THREAD

**Issue by EasyG0ing1** [SaptarshiSarkar12/Drifty#411]
### Describe the solution you'd like.

This is not so much a feature request as it is an idea for internally restructuring Drifty so that it no longer relies on text files for things like URL lists, download queues, or even settings (where leveraging the Java Preferences class has limitations). 

Instead, I am proposing that Drifty might be a good candidate to leverage a SQLite database for the storage and retrieval of dynamic data. The database / SQL method of storing and retrieving information is well understood and is relatively easy to implement.

A core class called something like DB could be created which contains all of the methods required to interact with external data. The database could have various tables, for example, a Jobs table which contains all of the relevant data for a download job in each record, or even a Log table which can store the various logs that Drifty generates. A Settings table could be used to store things like download folders, video quality preferences, or settings for GUI themes in the future. We could even keep a table for download statistics that keeps track of download metrics such as transfer speeds for past downloads etc. which then provides metrics that might be interesting in different contexts.

This change to Drifty could be developed non-invasively at first where it would just be a class that maintains the database file where the methods are defined then once they are in place and tested and working, it would then be a matter of converting those places throughout the program where it keeps information in flat files, and convert those methods to use the DB methods.

The database methods could even be presented to other developers as sort of an interface to external data where future development might be streamlined by the presence of the DB class, but this will be highly dependent on the nature of any future development.

**Comment by SaptarshiSarkar12** [comment]
Thank you for raising the issue @EasyG0ing1! It's a great improvement that can be implemented.
Would you like to work on this issue?

**Comment by Viereckig** [comment]
Hello @SaptarshiSarkar12. Are you still working on this or can i try it?

**Comment by SaptarshiSarkar12** [comment]
@Viereckig Thank you for your interest :smile:.
Sorry, due to lack of time, I could not focus on this issue.
Hence, you can work on this issue. If you face any problem, you can post it here and we can discuss on this issue in [our Discord Server](https://discord.gg/DeT4jXPfkG) :smiley:.

**Comment by Viereckig** [comment]
which Java SDK do i need to use?

**Comment by SaptarshiSarkar12** [comment]
@Viereckig The latest LTS Release - **JDK 21**.

**Comment by Viereckig** [comment]
Hey @SaptarshiSarkar12 have you ever encountered a problem when trying to run the program via Docker, as described in the README, where it fails to fetch packages?
![image](https://github.com/SaptarshiSarkar12/Drifty/assets/49343060/4c8cca07-d253-4411-a4d5-12a56c74980f)
![image](https://github.com/SaptarshiSarkar12/Drifty/assets/49343060/390f8715-e415-4d99-90e8-98dea609f1ca)

**Comment by SaptarshiSarkar12** [comment]
@Viereckig No, I haven't encountered such problems before. But, you can try running `sudo docker compose run cli` command again after deleting the cached images using `sudo docker system prune`. Please let me know if it works for you :smile:.

I tried today morning (IST timezone) and it works for me :point_down: 
![image](https://github.com/SaptarshiSarkar12/Drifty/assets/105960032/a59114ab-7689-45d7-b9c6-3f30cf338037)

**Comment by SaptarshiSarkar12** [comment]
@Viereckig If that does not work for you, you can try changing your DNS to `1.1.1.1` which is a *free public domain* offered by **Cloudflare**. You can change your DNS by following this tutorial - **https://one.one.one.one/dns/#setup-instructions**.

**Comment by Viereckig** [comment]
> @Viereckig If that does not work for you, you can try changing your DNS to `1.1.1.1` which is a _free public domain_ offered by **Cloudflare**. You can change your DNS by following this tutorial - **https://one.one.one.one/dns/#setup-instructions**.

already did that...

**Comment by Viereckig** [comment]
> @Viereckig No, I haven't encountered such problems before. But, you can try running `sudo docker compose run cli` command again after deleting the cached images using `sudo docker system prune`. Please let me know if it works for you 😄.
> 
> I tried today morning (IST timezone) and it works for me 👇 ![image](https://private-user-images.githubusercontent.com/105960032/321851206-a59114ab-7689-45d7-b9c6-3f30cf338037.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI5MDQxMjMsIm5iZiI6MTcxMjkwMzgyMywicGF0aCI6Ii8xMDU5NjAwMzIvMzIxODUxMjA2LWE1OTExNGFiLTc2ODktNDVkNy1iOWM2LTNmMzBjZjMzODAzNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMlQwNjM3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YmY0NjllMzAwZTY0OTFjYWI0MjJjMDEwYjY4ZDc1OTVkN2EzZGQyYmQ2YjVkYTM4YzAwNzkzNGY4OTMwYjBlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LM9LZ2bA6u0HjhbRl0GSBoDhJ30E5yMG9KZA4ZtjUQA)

yeah i try that thanks

**Comment by SaptarshiSarkar12** [comment]
> > @Viereckig No, I haven't encountered such problems before. But, you can try running `sudo docker compose run cli` command again after deleting the cached images using `sudo docker system prune`. Please let me know if it works for you 😄.
> > I tried today morning (IST timezone) and it works for me 👇 ![image](https://private-user-images.githubusercontent.com/105960032/321851206-a59114ab-7689-45d7-b9c6-3f30cf338037.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI5MDQxMjMsIm5iZiI6MTcxMjkwMzgyMywicGF0aCI6Ii8xMDU5NjAwMzIvMzIxODUxMjA2LWE1OTExNGFiLTc2ODktNDVkNy1iOWM2LTNmMzBjZjMzODAzNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMlQwNjM3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YmY0NjllMzAwZTY0OTFjYWI0MjJjMDEwYjY4ZDc1OTVkN2EzZGQyYmQ2YjVkYTM4YzAwNzkzNGY4OTMwYjBlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LM9LZ2bA6u0HjhbRl0GSBoDhJ30E5yMG9KZA4ZtjUQA)
> 
> yeah i try that thanks

@Viereckig Did that work?

**Comment by Viereckig** [comment]
> docker system prune

@SaptarshiSarkar12 sorry for the late response, I was very bussy the last few days! yeah it works now thanks

**Comment by SaptarshiSarkar12** [comment]
@Viereckig Are you facing any issues? Have you started working on this? Please let me know :smile:.

**Comment by SaptarshiSarkar12** [comment]
@Viereckig has informed me that he is no longer going to work on this issue.
This issue is now open for contribution.

**Comment by michalryn** [comment]
Hello @SaptarshiSarkar12, recently a friend of mine reached out to you, that we would like to work on this issue as a team. So as you asked, I'm leaving a comment 🙂.

**Comment by SaptarshiSarkar12** [comment]
@michalryn Hi Michał 👋!
Yeah, Kamal recently reached out to me regarding your team's work on this issue 😃.
I am assigning you as a representative of your team. Feel free to comment here if you face any problem 😀.

**Comment by bdaf** [comment]
Hey, I am Kamil* small mistake:D leaving a comment too.

**Comment by SaptarshiSarkar12** [comment]
@bdaf Hi Kamil :wave:!
I apologise for the mistake in writing your name :pray:.
Thank you for posting the comment :pray:. It really helps to keep track of all the people behind a single PR :smile: (as I believe Open-Source is first about the people, then about the code :upside_down_face:).
All the best to your team :+1:. I would love to see your team's work on this :heart:.

**Comment by Rafalrrr159** [comment]
As you asked, I also leave a comment :)

**Comment by SaptarshiSarkar12** [comment]
> As you asked, I also leave a comment :)

Thank you @Rafalrrr159 :smile:!
I am assigning you also as you are also a part of the team :)
All the best to you :+1:!
